applications/infra.py

```diff
@@ -0,0 +1,241 @@
+assert np.abs(sum(consumption_per_day) - sum(consumption_per_ts)) < 0.0001
+assert np.abs(sum(hydro_prod_per_ts_per_dam) - sum(hydro_prod_per_ts)) < 0.0001
+print(len(hydro_prod_per_ts), ' should be divided by 24')
+hydro_prod_per_day = block24(hydro_prod_per_ts)
+assert np.abs(sum(hydro_prod_per_ts_per_dam) - sum(hydro_prod_per_day)) < 0.0001
+assert len(hydro_prod_per_day) == len(consumption_per_day)
+print('daily hydro prod=', hydro_prod_per_day[:5])
+print('daily consumption=', consumption_per_day[:5])
+ax.plot(np.linspace(1,365,len(consumption_per_day)), consumption_per_day, label='consumption')
+if True:
+    ax.plot(np.linspace(1,365,len(hydro_prod_per_day)), hydro_prod_per_day, label='hydro')
+    assert sum(consumption_per_day) >= sum(hydro_prod_per_ts_per_dam)
+    assert len(hydro_prod_per_ts_per_dam) % N == 0
+    assert len(hydro_prod_per_ts_per_dam) % (N * 24) == 0
+    assert len(hydro_prod_per_ts_per_dam) % (N * 24 * 365) == 0
+    assert abs(sum(hydro_prod_per_ts_per_dam) - sum(hydro_prod_per_ts)) < 1e-4
+    assert sum(hydro_prod_per_ts) <= sum(consumption_per_ts)
+    #hydro_ts = [0.] * (len(hydro_prod_per_ts_per_dam) // N)
+    #for j in range(len(hydro_prod_per_ts_per_dam)):
+    #    hydro_ts[j % N] += hydro_prod_per_ts_per_dam[j]
+    for i in range(min(N, 3)):
+        # Let us consider a specific dam
+        hydro_ts = [hydro_prod_per_ts_per_dam[j] for j
+                in range(i, len(hydro_prod_per_ts_per_dam), N)]
+        hydro_day = block24(hydro_ts)
+        ax.plot(np.linspace(1,365,len(hydro_day)), hydro_day, label='dam ' + str(i) + ' prod')
+ax.set_xlabel('time step')
+ax.set_ylabel('production per day')
+ax.legend(loc='best')
+
+# Plot consumption per hour of the day and decomposition of production.
+ax = plt.subplot(2, 2, 3)
+def deblock24(x):
+    result = [0] * 24
+    for i in range(0, len(x)):
+        result[i % 24] += x[i]
+    assert len(result) == 24
+    return result
+        
+consumption_per_hour = deblock24(consumption_per_ts)
+print(consumption_per_hour)
+hydro_prod_per_hour = deblock24(hydro_prod_per_ts)
+print(hydro_prod_per_hour)
+assert len(consumption_per_hour) == 24
+ax.plot(np.linspace(1,24,len(consumption_per_hour)), consumption_per_hour, label='consumption')
+ax.plot(np.linspace(1,24,len(hydro_prod_per_hour)), hydro_prod_per_hour, label='hydro')
+ax.plot(np.linspace(1,24,len(hydro_prod_per_hour)), [a-b for a, b in zip(consumption_per_hour, hydro_prod_per_hour)], label='thermal')
+for i in range(min(N,3)):
+    hydro_ts = [hydro_prod_per_ts_per_dam[j] for j in range(i, len(hydro_prod_per_ts), N)]
+    hydro_hour = deblock24(hydro_ts)
+    ax.plot(np.linspace(1,24,len(hydro_hour)), hydro_hour, label='dam ' + str(i) + ' prod')
+ax.set_xlabel('time step')
+ax.set_ylabel('production per hour')
+ax.legend(loc='best')
+plt.savefig("ps.png")
+plt.show()
+plt.waitforbuttonpress()
+recommendation = optimizer.recommend()  #minimize(square)
+print(recommendation)  # optimal args and kwargs
```

Comment:
This stuff is mostly duplicated (very similar to powersystems/core.py). Keep only one or it will be very confusing.

Reply:
Done.